src/tools/rustfmt/src/types.rs

```diff
@@ -552,6 +552,8 @@ impl Rewrite for ast::GenericBound {
                     ast::TraitBoundModifier::MaybeConstMaybe => poly_trait_ref
                         .rewrite(context, shape.offset_left(8)?)
                         .map(|s| format!("~const ?{}", s)),
+                    rustc_ast::TraitBoundModifier::Negative
+                    | rustc_ast::TraitBoundModifier::MaybeConstNegative => None,
```

Comment:
Not formatting this at all.

Reply:
hmm, i'm worried this may actually result in rustfmt nuking these modifiers. would encourage adding a test in some file under `src/tools/rustfmt/tests/target/` (existing is fine if you see one that seems relevant, new file is fine too) that includes a snippet with the new modifiers just to make sure.

Reply:
actually sorry. i don't think it would end up eliding them, but i do think it could result in formatting failure that could cascade up quite a ways. i'd suggest the below instead

```suggestion
                    rustc_ast::TraitBoundModifier::Negative => poly_trait_ref
                         .rewrite(context, shape.offset_left(1)?)
                         .map(|s| format!("!{}", s)),
                     rustc_ast::TraitBoundModifier::MaybeConstNegative => poly_trait_ref
                         .rewrite(context, shape.offset_left(8)?)
                         .map(|s| format!("~const !{}", s)),
```

(disclaimer: mostly hand written in github ui browser so typo and formatting issues possible)

Reply:
Ok, I'll apply that change modulo compiler errors.